skbio/sequence/tests/test_sequence.py

```diff
@@ -85,6 +85,11 @@ def test_concat_on_subclass(self):
         self.assertIs(type(result), SequenceSubclass)
         self.assertEqual(result, SequenceSubclass("123123"))
 
+    def test_concat_on_empty_iterator(self):
+        result = SequenceSubclass.concat((_ for _ in []))
```

Comment:
numpy wouldn't always have a dtype. I think we can avoid that because the output is always predictable being a classmethod instead of a function. 

I also added this behaviour because it makes the implementation simpler as I don't need to special case an empty slice situation when pulling out a column.

If we feel strongly about raising an error that is fine, but I think that 
`Sequence.concat([]) -> Sequence("")` is a logically consistent result.